### PR TITLE
some timestamp refinements

### PR DESCRIPTION
**Pull request by Goober5000**
1. Make sure the legacy `timestamp_until()` and `timestamp_since()` functions check for any type of invalid timestamp
2. Add a useful `isFinite()` function
3. A timestamp delta on an immediate timestamp is more properly the actual delta, rather than another immediate timestamp

### CODE REVIEW

**Review by wookieejedi — APPROVED** [review]
Tests work as expected for me and is straightforward PR. Good to see `isFinite()` being in both TIMESTAMP and `UI_TIMESTAMP`, too.

File: code/io/timer.cpp [hunk 1]
```diff
@@ -180,17 +180,23 @@ UI_TIMESTAMP ui_timestamp() {
 
 TIMESTAMP timestamp_delta(TIMESTAMP stamp, int delta_ms)
 {
-	if (!stamp.isValid() || stamp.isImmediate() || stamp.isNever())
+	if (!stamp.isValid() || stamp.isNever())
```
Comment (MageKing17) [inline code/io/timer.cpp:183]: This could be replaced with the new function:
```suggestion
	if (!stamp.isFinite())
```

File: code/io/timer.cpp [hunk 2]
```diff
@@ -180,17 +180,23 @@ UI_TIMESTAMP ui_timestamp() {
...
+	if (stamp.isImmediate())
+		return TIMESTAMP(timestamp_ms() + delta_ms);
+
 	return TIMESTAMP(stamp.value() + delta_ms);
 }
 
 UI_TIMESTAMP ui_timestamp_delta(UI_TIMESTAMP stamp, int delta_ms)
 {
-	if (!stamp.isValid() || stamp.isImmediate() || stamp.isNever())
+	if (!stamp.isValid() || stamp.isNever())
```
Comment (MageKing17) [inline code/io/timer.cpp:194]: Same here.
```suggestion
	if (!stamp.isFinite())
```

**Review by MageKing17 — COMMENTED** [review]
The new `isFinite()` function could immediately be put to use simplifying conditionals.

File: code/io/timer.cpp (hunk 2 above)
Comment (Goober5000) [reply code/io/timer.cpp]: Also done

**Pull request merged**